src/Disks/DiskRestartProxy.h

```diff
@@ -64,7 +64,8 @@ class DiskRestartProxy : public DiskDecorator
     void truncateFile(const String & path, size_t size) override;
     String getUniqueId(const String & path) const override;
     bool checkUniqueId(const String & id) const override;
-    String getCacheBasePath() const override;
+
+    const String & getCacheBasePath() const override;
```

Comment:
Because now there is either such non-empty path from an existing object or exception, so reference is ok. Previously, reference was impossible as empty string could be returned.